File: history/0.1.2/tatala-java/src/com/qileyuan/tatala/socket/to/helper/LongArrayMapHelper.java
package com.qileyuan.tatala.socket.to.helper;

import java.util.Iterator;
import java.util.Map;

import com.qileyuan.tatala.socket.io.TransferOutputStream;
import com.qileyuan.tatala.socket.to.TransferObject;
import com.qileyuan.tatala.socket.util.TransferUtil;

public class LongArrayMapHelper {

	public static int getByteArrayLength(Map<String, long[]> longArrayMap){
		int blength = 0;
		Iterator<String> keyIt = longArrayMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			blength += TransferUtil.getLengthOfByte();
			// key
			String key = (String) keyIt.next();
			blength += TransferUtil.getLengthOfString(key);
			// value
			long[] values = longArrayMap.get(key);
			blength += TransferUtil.getLengthOfLongArray(values);
		}
		return blength;
	}
	
	public static void getByteArray(Map<String, long[]> longArrayMap, TransferOutputStream touts) {
		Iterator<String> keyIt = longArrayMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			touts.writeByte(TransferObject.DATATYPE_LONGARRAY);
			// key
			String key = (String) keyIt.next();
			touts.writeString(key);
			// value
			long[] values = longArrayMap.get(key);
			touts.writeLongArray(values);
		}
	}
}

File: history/0.1.2/tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/helper/LongArrayMapHelper.cs
using System;
using System.Collections.Generic;
using QiLeYuan.Tatala.socket.io;
using QiLeYuan.Tatala.socket.util;

namespace QiLeYuan.Tatala.socket.to.helper {

    public class LongArrayMapHelper {
        public static int getByteArrayLength(Dictionary<String, long[]> longArrayMap) {
            int blength = 0;
            foreach (KeyValuePair<String, long[]> item in longArrayMap) {
                // type
                blength += TransferUtil.getLengthOfByte();
                // key
                String key = item.Key;
                blength += TransferUtil.getLengthOfString(key);
                // value
                long[] values = item.Value;
                blength += TransferUtil.getLengthOfLongArray(values);
            }
            return blength;
        }

        public static void getByteArray(Dictionary<String, long[]> longArrayMap, TransferOutputStream touts) {
            foreach (KeyValuePair<String, long[]> item in longArrayMap) {
                // type
                touts.writeByte(TransferObject.DATATYPE_LONGARRAY);
                // key
                String key = item.Key;
                touts.writeString(key);
                // value
                long[] values = item.Value;
                touts.writeLongArray(values);
            }
        }
    }
}